feat: update readme (#75)
adorsys/status-list-server: README.md

- **Response:**
  - `200 OK`: Server is running.
  
### Register Issuer
- **Endpoint**: `POST /credentials/`
- **Description**: Allows issuers to register their public key and identifier for later authentication
- **Request Body**
  ```json
  {
    "issuer": "<issuer_id>",
    "public_key": "<public_key.pem>",
    "alg": "ES256"
  }
  ```
  - `issuer`: Unique identifier for the issuer
  - `public_key`: PEM-encoded public key in base64 format
  - `alg`: Must be "ES256" (ECDSA with P-256 and SHA-256)
 
### Publish Status List
- **Endpoint**: `POST /statuslists/publish` 
- **Description**: Allows an issuer to publish their token status list

> - **Authorization**: Requires a valid JWT token with the issuer's ID as the `kid` (Key ID) in the header

Reviewer: ```suggestion
- **Authorization**: Requires a valid signed JWT token with the corresponding registered private key with the issuer's ID as the `kid` (Key ID) in the header
```

- **Request Body**

>   ```json
>   [
>       { "index": 1, "status": "INVALID" },
>       { "index": 8, "status": "VALID" }
>   ]
>   ```

Reviewer: ```suggestion
  ```json
  {"list_id": "30202cc6-1e3f-4479-a567-74e86ad73693",
  [
      { "index": 1, "status": "INVALID" },
      { "index": 8, "status": "VALID" }
  ]
  }
  ```
```

  - `index`: Position in the status list
  - `status`: Status value (VALID, INVALID, SUSPENDED, APPLICATIONSPECIFIC)

### Update Status List

- **Endpoint:** `PUT /statuslists/{list_id}`
- **Description:** Allows an issuer to update an existing status list

> - **Authorization:** Requires a valid JWT token with the issuer's ID as the `kid` in the header

Reviewer: ```suggestion
- **Authorization:** Requires a valid signed JWT token with the corresponding registered private key with the issuer's ID as the `kid` in the header
```

  }
  ```
  
  - `list_id`: UUID of the status list to update

>   - `bits`: Number of bits per status entry

Reviewer: ```suggestion

```

>   - `updates`: Array of status updates

Reviewer: Provide and example of a status update payload

    - `index`: Position in the status list
    - `status`: New status value (VALID, INVALID, SUSPENDED, APPLICATIONSPECIFIC)

- **Responses:**
  - `200 OK`: Update successful
  - `400 BAD REQUEST`: Invalid input data
  - `401 UNAUTHORIZED`: Invalid or missing JWT token
  - `403 FORBIDDEN`: Token issuer doesn't match list owner

### Retrieve Status List

- **Endpoint:** `GET /statuslists/{list_id}`

> - **Description:** Retrieves the current status list. This endpoint is publicly accessible with no authentication required.

Reviewer: ```suggestion
- **Description:** Retrieves the status list for the request list_id. This endpoint is publicly accessible with no authentication required.
```

> - **Responses:**

Reviewer: You did not talk about the content type of the request if we are requesting for the status list in `jwt` or `cwt`

  - `200 OK`: Returns the status list in a compressed and encoded format
  - `404 NOT FOUND`: Status list not found

## Authentication

The server uses JWT-based authentication with the following requirements:

1. Issuers must first register their public key using the `/credentials/` endpoint
2. All authenticated requests must include a JWT token in the Authorization header:
   ```
   Authorization: Bearer <jwt_token>
   ```
3. The JWT token must:

>    - Be signed with ES256 (ECDSA with P-256 and SHA-256)

Reviewer: must it be signed with ES256 ?
